### PR TITLE
Replace date time patch with refinements

### PR DESCRIPTION
**Pull request by andywenk**
Added refinements and removed monkey patches

Up from Ruby 2.0 there is support for refinements. This is a better way to redefine or add functionality to existing (core) classes. The changes here are motivated due to the error, that the patches for Date and Time have breaking influences to other gems. 

The folder core_ext still exists with the two patched classes Date and Time. Because this change will require Ruby >= 2.0, users with older projects and Ruby versions have still the possibility to remove the refinements by not 'using' them in module CouchPotato. This will be described in the README in more detail.

### CODE REVIEW

File: lib/refinements/time.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    return nil if string.nil?
+    d = DateTime.parse(string.to_s).new_offset
+    self.utc(d.year, d.month, d.day, d.hour, d.min, d.sec).in_time_zone
+  end
+end
+
+module CouchPotatoRefinements
+  refine Time do
+    def to_json(*a)
+    %("#{as_json}")
```
Comment (langalex) [inline lib/refinements/time.rb]: indentation broken

File: README.md [hunk 2]
```diff
@@ -39,6 +39,23 @@ Couch Potato is hosted as a gem which you can install like this:
 
     gem install couch_potato
 
+### Now using Ruby refinements
```
Comment (langalex) [inline README.md]: I would leave this out of the Readme and put something in changes.md instead. Also, remove `core_ext`. Projects using < 2.0 just have to use old versions of Couch Potato.

File: lib/refinements/time.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    end
+
+    def as_json(*args)
+      getutc.strftime("%Y/%m/%d %H:%M:%S +0000")
+    end
+  end
+end
+
+ActiveSupport::TimeWithZone.class_eval do
```
Comment (langalex) [inline lib/refinements/time.rb]: so here we are still doing some evil monkey patching aren't we? is this necessary?

File: lib/refinements/time.rb (hunk 1 above)
Comment (andywenk) [inline lib/refinements/time.rb]: oh sorry my oversight. will fix

File: lib/refinements/time.rb (hunk 3 above)
Comment (andywenk) [inline lib/refinements/time.rb]: hm ... you're right - have to check this

File: README.md (hunk 2 above)
Comment (andywenk) [inline README.md]: ok - no prob. But don't you think at least an info about this should be added. Like 

'When using an older version than Ruby 2.0 please use CouchPotato 1.4.0' 

?

Comment (langalex) [inline README.md]: ok :)

**Comment by andywenk** [comment]
reworked the PR

**Comment by langalex** [comment]
looks like we broke 2.0, too. so, can you change it to require at least 2.1 (and also remove < 2.1 from travis.yml). if it's too much work for you i can also take over from here.

**Comment by andywenk** [comment]
Hm ok. refinements are marked as experimental in 2.0 I saw. So I should have tested against 2.0.

I started to change it and then installed rbx-2.2.10 to let the tests run. refinements are not supported there. So I tried to install rbx-2.5.2 what failed miserably. I had to configure it with (and install llvm before with brew):

    rvm install rbx -C --llvm-config=/usr/local/Cellar/llvm/3.5.1/bin/llvm-config

This just as an info for others. 
BUT - refinements are not available in Rubinius 2.5.2 also.

So it's your decision what to do now. Drop Rubinius what would mean we only support Ruby MRI up from 2.1 fro now.

**Comment by andywenk** [comment]
this is going to be a tough one it seems ;-). I just realised by running the tests in our Rails Applikation, that there are failing tests without the `still bad ActiveSupport::TimeWithZone.class_eval` inside `time.rb`. Will investigate further and report.

**Comment by andywenk** [comment]
ok all my tests show that this is needed:

    ActiveSupport::TimeWithZone.class_eval do
      using CouchPotatoRefinements

      def as_json(*args)
        utc.as_json
      end
    end

Otherwise CouchPotato is not working properly in a Rails application. Unfortunately, I was not able to put this also in a refinement. I think before we go further, some decisions have to be made. I'll wait now for your ideas ;-).

**Comment by langalex** [comment]
this was fixed in 4e519fe0fefacfe5826f2f4e9cbcddbc444c4f3e, without using refinements or anything.

**Comment by andywenk** [comment]
cool! Thanks

**Pull request closed without merging**